invenio_records_resources/resources/files/resource.py

```diff
@@ -180,6 +182,12 @@ def read_content(self):
         if item is None:
             abort(404)
 
+        # emit file download stats event
+        obj = item._file.object_version
+        emitter = current_stats.get_event_emitter("file-download")
+        if obj is not None and emitter is not None:
+            emitter(current_app, record=item._record, obj=obj, via_api=True)
```

Comment:
minor: sending an underscored variable seems to defy the purpose of having it encapsulated. Also `obj` is a bit heavy to be sent? I'm wondering if we can send e.g. files.to_dict() and some sort of serialization of `obj`. 

I seem to recall payloads were not to be discussed now? if so, just ignore my comment ❌  😂

Reply:
bit of context:
the arguments to `emitter(...)` aren't sent directly as part of a message to the MQ –  rather, they are passed to the registered **event builders** that use them to create a dictionary populated with data fetched from the passed args/kwargs.
after all registered event builders have added their parts, the constructed **event** dictionary is sent off via the MQ.
in the case of the `file-download` events, this event builder will only extract a few properties from the `obj`: https://github.com/inveniosoftware/invenio-app-rdm/pull/2080/files#diff-b4d96b23efeba4170d5d72d1330d7f17d19ebfb5e07d211939bfd720543dbb66R23-R50
the default event builders for InvenioRDM can be found in the app-rdm config: https://github.com/inveniosoftware/invenio-app-rdm/pull/2080/files#diff-988eb51a0bc0a8bf6efd0b04d1bb6b6d8f24cc80504ea36699a202904a20af4dR876-R879

**but yes**, good point!
we could probably send some version of `record.to_dict()` and `obj.to_dict()` to the event builders rather than the actual objects themselves... that would probably help out a lot in the future when instead of building & emitting events synchronously like here, we'd add a (probably MQ-based) event bus :smile:

Reply:
That makes, sense. In my head it would make more sense not to access an `_` value. However, more sense would make an event bus. So for me is fine if left like this since it will be re-worked (hopefully) at some point. So up to you. Thanks a lot for the explanation!